### PR TITLE
Add login page

### PR DESCRIPTION
**Pull request by HuiminFrontend**

### CODE REVIEW

File: src/hooks/useStore_login.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    set(state => ({
+      login: {
+        token: state.login.token,
+        credentials: {...state.login.credentials, username: username, password: password},
+      },
+    }));
+  },
+});
+
+const useStoreLogin = create(persist(loginStore, {name: 'login'}));
```
Comment (JessicaLoers) [inline src/hooks/useStore_login.js:22]: Just a picky note: if you rename the useStore,  it's a common pattern zu name it something like `useLogin` or `useLoginStore`. 
Also, I would like to suggest that you name your file the same as your store. e.g. `userLogin.js` or `userLoginStore.js`

File: .env.example [hunk 2]
```diff
@@ -1 +0,0 @@
-MONGODB_URI=mongodb://localhost:27017/todos
```
Comment (JessicaLoers) [inline .env.example:1]: I would keep the `.env.example` as a guide which contains an example of your `MONGODB_URI`

File: src/pages/Login.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+import {useState} from 'react';
+import {AiFillTwitterCircle, AiFillGoogleCircle} from 'react-icons/ai';
+import {RiFacebookCircleFill} from 'react-icons/ri';
+import {useNavigate} from 'react-router-dom';
+import styled from 'styled-components';
+
+import useStoreLogin from '../hooks/useStore_login.js';
```
Comment (JessicaLoers) [inline src/pages/Login.js:7]: If you change the name of the `useStore `according to my suggestion, please keep in mind to change it here too.🙂

File: src/pages/Login.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+export default function Login() {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [inputCredentials, setInputCredentials] = useState({
+    username: '',
+    password: '',
+  });
+
+  const {setToken, setCredentials} = useStoreLogin(state => ({
```
Comment (JessicaLoers) [inline src/pages/Login.js:16]: Please see my comment above.

File: src/pages/Login.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        <Title>Login</Title>
+        <Username>
+          Username
+          <Input type="text" name="username" value={inputCredentials.username} onChange={handleChange} />
+        </Username>
+        <Password>
+          Password
+          <Input type="password" name="password" value={inputCredentials.password} onChange={handleChange} />
+        </Password>
+        {errorMessage && <div style={{color: 'crimson'}}>{errorMessage}</div>}
```
Comment (JessicaLoers) [inline src/pages/Login.js:65]: I would recommend you not to use inline styles.

File: src/pages/Login.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        <LoginButton>LOGIN</LoginButton>
+        <SignUpTipps>
+          Or sign up Using
+          <Logos>
+            <RiFacebookCircleFill style={{color: '#395598'}} />
```
Comment (JessicaLoers) [inline src/pages/Login.js:70]: Please see my comment above 🧐

File: src/pages/Login.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+  background-color: white;
+  border-radius: 8px;
+  padding: 2rem;
+`;
+
+const Title = styled.h1`
+  margin: 0 auto;
```
Comment (JessicaLoers) [inline src/pages/Login.js:101]: Just an other picky note: to center the `<h1>` element you can use `text-align: center`; as well.

File: src/pages/Orders.js [hunk 8]
```diff
@@ -1,44 +1,70 @@
-import styled from 'styled-components';
+import styled, {css} from 'styled-components';
 
 import Header from '../components/Header/Header.js';
 import Navigation from '../components/Navigation/Navigation';
+import useStoreLogin from '../hooks/useStore_login';
```
Comment (JessicaLoers) [inline src/pages/Orders.js:5]: If you change the name of the useStore according to my suggestion, please keep in mind to change it here too.🙂

File: src/pages/Orders.js [hunk 9]
```diff
@@ -1,44 +1,70 @@
...
 import useStoreOrder from '../hooks/useStore_order';
 
 export default function Orders() {
   const {orders} = useStoreOrder(state => ({
     orders: state.orders,
   }));
 
+  const {login, setToken} = useStoreLogin(state => ({
```
Comment (JessicaLoers) [inline src/pages/Orders.js:13]: See my comment above.

File: src/pages/Orders.js [hunk 10]
```diff
@@ -1,44 +1,70 @@
...
+          <Greet>
+            <span>Hello:</span>
+            <span>{login.credentials.username}</span>
+          </Greet>
+          <LogoutButton onClick={logout}>Logout</LogoutButton>
+        </UserComponent>
+      ) : (
+        <>
+          <SpeechBubbleTipps>
+            You have not logged in. After login, you can receive messages of coupons and new dishes, reserve tables,
```
Comment (JessicaLoers) [inline src/pages/Orders.js:36]: I would wrap a more specific and semantic element around the notifications text e.g. `<p>` ... you might even come up with something smarter than my suggestion 🫣

**Review by JessicaLoers — APPROVED** [review]
**Hi @Huimin66** 
really, really good job 🚀

I have just some minor notes, suggestions and comments.

File: src/hooks/useStore_login.js (hunk 1 above)
Comment (HuiminFrontend) [reply src/hooks/useStore_login.js:22]: Thank you for suggestion, I'll change it.

File: src/pages/Login.js (hunk 3 above)
Comment (HuiminFrontend) [reply src/pages/Login.js:7]: Ok

File: src/pages/Login.js (hunk 7 above)
Comment (HuiminFrontend) [reply src/pages/Login.js:101]: I'll try it out. Thank you, good to know it works with different codes.

File: .env.example (hunk 2 above)
Comment (HuiminFrontend) [reply .env.example:1]: I can do it.

**Pull request merged**